### ISSUE TITLE
What version of zig should this compile with?

### ISSUE THREAD

**Issue by pmwhite** [donpdonp/zootdeck#1]
I tried to build this with the latest released version of zig (0.4.0), but it did not compile due to the restructuring of the modules which `makeDir` calls. I then tried to build with the most recent master of the zig repository, but this also didn't compile. Perhaps it would be good to keep up to date on the README what revision number of the repository you expect this to build with.

**Comment by donpdonp** [comment]
an excellent question. the zig on my laptop that I use to develop zootdeck is
```
$ zig version
0.4.0+7f23dac6d
```

When last I tried zig HEAD, zootdeck would crash the compiler, but that was a few weeks ago. I need to try it with a more recent HEAD version.

**Comment by a-palchikov** [comment]
There seems to be either a missing build step or a missing pre-compiled `ragel/lang.c`:
```
$ zig build
Unable to hash /path/to/zig/github.com/donpdonp/zootdeck/ragel/lang.c: file not found
zootdeck...The following command exited with error code 1:
...
```
Any help would be appreciated!

**Comment by donpdonp** [comment]
if you use `make` there is a rule to build lang.c using ragel.

```
turnip projects/zootdeck$ rm ragel/lang.c
turnip projects/zootdeck$ make
ragel -o ragel/lang.c ragel/lang.c.rl
zig build
./src/gui.zig:28:5: error: extern functions have no body
pub extern fn go(data: ?*c_void) ?*c_void {
```

note that master currenly fails to build as the code needs to be updated for zig 0.6 and I havent put much time into zootdeck lately. thanks for the interest though!

**Comment by a-palchikov** [comment]
Did not notice Makefile for some reason - thanks for clearing that up.

**Comment by donpdonp** [comment]
If you still want to give this a try, i just pushed the last fix into master for zig head (zig 0.6 is already old :open_mouth:

**Comment by donpdonp** [comment]
zootdeck compiles under the latest zig 0.10.0-dev.2624+d506275a0
